Zend/zend_generators.c

```diff
@@ -163,6 +163,52 @@ ZEND_API void zend_generator_close(zend_generator *generator, zend_bool finished
 
 static zend_generator *zend_generator_get_child(zend_generator_node *node, zend_generator *leaf);
 
+static void zend_generator_update_leaf_of_child(zend_generator_node *node, zend_generator *from_leaf, zend_generator *to_leaf)
+{
+	if (node->ptr.leaf == from_leaf) {
+		node->ptr.leaf = to_leaf;
+	}
+	if (node->children <= 1) {
+		node->child.single.leaf = to_leaf;
+	} else {
+		HashTable *ht = node->child.ht;
+		zend_generator *child = zend_hash_index_find_ptr(ht, (zend_ulong) from_leaf);
+		if (child) {
+			zend_hash_index_del(ht, (zend_ulong) from_leaf);
+			zend_hash_index_add_ptr(ht, (zend_ulong) to_leaf, child);
+		}
+	}
+}
+
+static void zend_generator_remove_leaf_child(zend_generator_node *node, zend_generator *leaf, zend_generator *replace_leaf) {
+	if (node->children > 1) {
```

Comment:
Shouldn't we have `ZEND_ASSERT(node->children > 1)` here? We only get here if `parent.node->children > 1`, which also implies that `root.node->children > 1`, I believe, as we store all the leaves.

Or else this is missing test coverage for the case where node->children <= 1 here...

Reply:
No, it does not imply that, we only store all the leaves if the node has multiple direct children (that's why this multi_children_node searching needs to exist, winding down the leafs until the first node with multiple children is found),

Reply:
Okay, I see. It would be great to have a test case that hits the else part, as none of the existing ones do.